flaml/searcher/blendsearch.py

```diff
@@ -734,6 +736,20 @@ def suggest(self, trial_id: str) -> Optional[Dict]:
                 result = {self._metric: reward, self.cost_attr: 1, "config": config}
                 self.on_trial_complete(trial_id, result)
                 return None
+
+        if self._use_incumbent_result:
+            thread_obj = [
+                thread.obj_best1
+                for id, thread in self._search_thread_pool.items()
+                if id and thread._is_ls
+            ]
+            if len(thread_obj) != 0:
+                results = [
+                    thread.best_result
+                    for id, thread in self._search_thread_pool.items()
+                    if id and thread._is_ls
+                ]
+                config["incumbent_result"] = results[np.argmin(thread_obj)]

```

Comment:
We don't want to pass the best incumbent among all threads.
We should only pass the incumbent for the corresponding thread that suggests this config.
For now, only do it for local search thread